src/certificate/commands/MsvaCertificateCommand.ts

```diff
@@ -0,0 +1,64 @@
+import moment from 'moment';
+import { Service } from 'typedi';
+import { DefectService } from '../../defect/DefectService';
+import { ITestResult } from '../../models';
+import { ICertificatePayload } from '../../models';
+import { IRequiredStandard } from '../../models';
+import { CERTIFICATE_DATA } from '../../models/Enums';
+import { ICertificatePayloadCommand } from '../ICertificatePayloadCommand';
+
+@Service()
+export class MsvaCertificateCommand implements ICertificatePayloadCommand {
+	private type?: CERTIFICATE_DATA;
+
+	constructor(private defectService: DefectService) {}
+
+	private certificateIsAnMsva = (): boolean => this.type === CERTIFICATE_DATA.MSVA_DATA;
+
+	initialise(type: CERTIFICATE_DATA, isWelsh = false) {
+		this.type = type;
+	}
+
+	public generate(testResult: ITestResult): Promise<ICertificatePayload> {
```

Comment:
For consistency is it possible to make this generate method async please, removing the need for the Promise.resolve() wrapper in this return and the return on line 44.

Reply:
Same as above :)